### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by kevmoo**
The runner changes are a bit nuts. Don't worry about them too much.

Changes to the core libraries, though, I want you to feel good about them.

### CODE REVIEW

File: bin/main.dart [hunk 1]
```diff
@@ -14,26 +14,52 @@ final _none = '\u001b[0m';
 String gray(text) => "$_gray$text$_none";
 
 main(List<String> arguments) async {
+  // as provided, `arguments` is fixed length – turn it into a mutable `List`
```
Comment (isoos) [inline bin/main.dart]: why not use the package:args?

File: bin/main.dart [hunk 2]
```diff
@@ -14,26 +14,52 @@ final _none = '\u001b[0m';
...
+
+  // Docker is WEIRD
+  // The SIGTERM signal sent to `docker run...` DOES propagate a signal to the
+  // running process. But...
+  //   * It is received as SIGINT
+  //   * It won't terminal the Dart process either – *BUT* we can listen for it
+  // So this is how we do "clean" shutdown when running in Docker.
+  var subscription = getSignals().listen((sig) async {
+    log.Logger.root.severe("Received signal `$sig` – terminating.");
+    exit(130);
```
Comment (isoos) [inline bin/main.dart]: out of curiosity: why 130? :)

File: bin/main.dart [hunk 3]
```diff
@@ -56,12 +82,33 @@ main(List<String> arguments) async {
...
       exitCode = 1;
     }
   } finally {
     tempDir.deleteSync(recursive: true);
   }
+
+  await subscription.cancel();
+}
+
+void _logWriter(record) {
```
Comment (isoos) [inline bin/main.dart]: nit: always wondered if there is a package for this? let's keep it for now, but I'd really like to know what others are using (and replace my similar hacks with that).

File: lib/src/pub_summary.dart [hunk 4]
```diff
@@ -134,27 +134,45 @@ class PubSummary {
...
       if (versionConstraint == null) {
-        stderr.writeln('No constraint provided for $package');
+        logWeird('BAD! No constraint provided for $package');
         return;
       }
 
       if (versionConstraint is Map) {
-        stderr.writeln(
-            'Non-versioned constraint for $package\n  $versionConstraint');
+        if (versionConstraint['sdk'] == 'flutter') {
```
Comment (isoos) [inline lib/src/pub_summary.dart]: is this a duplicate with `Pubspec.dependsOnFlutterSdk` (in `pubspec.dart`)?

File: bin/main.dart (hunk 2 above)
Comment (kevmoo) [inline bin/main.dart]: 128 + 2 = ~convention for user killed.

http://tldp.org/LDP/abs/html/exitcodes.html

File: bin/main.dart (hunk 3 above)
Comment (kevmoo) [inline bin/main.dart]: Agreed!

File: bin/main.dart (hunk 1 above)
Comment (kevmoo) [inline bin/main.dart]: I'm lazy! 😄 

I have it in there as a TODO - somewhere

File: lib/src/pub_summary.dart (hunk 4 above)
Comment (kevmoo) [inline lib/src/pub_summary.dart]: I'll look and see if I can fix

Comment (kevmoo) [inline lib/src/pub_summary.dart]: Ah! No. This is detecting if a package depends on flutter – not the whole Pubpsec/package.

**Pull request merged**